# Name: Veleña, Harold V.

<hr>Analyst: Harold V. Veleña</hr>

In [2]:
pip install plotly


[notice] A new release of pip is available: 24.3.1 -> 25.2
[notice] To update, run: python.exe -m pip install --upgrade pip


In [3]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import warnings
warnings.filterwarnings("ignore")

<p>Importing dataset to Jupyter notebook</p>
<p></p>

In [4]:
a = np.array([2.2, 3, 4])
a.dtype


dtype('float64')

# Getting the penguins data

In [5]:
penguins = pd.read_csv('penguins.csv')

penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

# How to create a Column

In [6]:
penguins["mass_kg"] = round(penguins["body_mass_g"] / 1000,1)
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  mass_kg  
0         3750.0    male  2007      3.8  
1         3800.0  female  2007      3.8  
2         3250.0  female  2007      3.2  
3            NaN     NaN  2007      NaN  
4         3450.0  female  2007      3.4  
..           ...     ...   ...      ...  
339       4000.0    male  2009      4.0  
340       3400.0  female  2009      3.4  
341       3775.0    male  2009      3.8  
342       4100.0    male  2009      4.1  
343       3775.0  female  2009      3.8  

[344 rows x 9 columns]

# Descriptive Analysis

<h2>Getting the Top 5</h2>

In [7]:
penguins.head()

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0  Adelie  Torgersen            39.1           18.7              181.0   
1  Adelie  Torgersen            39.5           17.4              186.0   
2  Adelie  Torgersen            40.3           18.0              195.0   
3  Adelie  Torgersen             NaN            NaN                NaN   
4  Adelie  Torgersen            36.7           19.3              193.0   

   body_mass_g     sex  year  mass_kg  
0       3750.0    male  2007      3.8  
1       3800.0  female  2007      3.8  
2       3250.0  female  2007      3.2  
3          NaN     NaN  2007      NaN  
4       3450.0  female  2007      3.4

In [8]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            344 non-null    object 
 1   island             344 non-null    object 
 2   bill_length_mm     342 non-null    float64
 3   bill_depth_mm      342 non-null    float64
 4   flipper_length_mm  342 non-null    float64
 5   body_mass_g        342 non-null    float64
 6   sex                333 non-null    object 
 7   year               344 non-null    int64  
 8   mass_kg            342 non-null    float64
dtypes: float64(5), int64(1), object(3)
memory usage: 24.3+ KB


In [9]:
penguins.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      342.000000     342.000000         342.000000   342.000000   
mean        43.921930      17.151170         200.915205  4201.754386   
std          5.459584       1.974793          14.061714   801.954536   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.225000      15.600000         190.000000  3550.000000   
50%         44.450000      17.300000         197.000000  4050.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year     mass_kg  
count   344.000000  342.000000  
mean   2008.029070    4.202047  
std       0.818356    0.803089  
min    2007.000000    2.700000  
25%    2007.000000    3.600000  
50%    2008.000000    4.000000  
75%    2009.000000    4.800000  
max    2009.000000    6.300000

In [10]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
mass_kg               2
dtype: int64

In [11]:
penguins[penguins["island"] == 'Dream']

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
30      Adelie  Dream            39.5           16.7              178.0   
31      Adelie  Dream            37.2           18.1              178.0   
32      Adelie  Dream            39.5           17.8              188.0   
33      Adelie  Dream            40.9           18.9              184.0   
34      Adelie  Dream            36.4           17.0              195.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  mass_kg  
30        3250.0  female  2007      3.2  
31        3900.0    male  2007      3.9  
32        3300.0  female  2007      3.3  
33        3900.0    male  2007      3.9  
34        3325.0  female  2007      3.3  
..           ...     ...   ...      ...  
339       4000.0    male  2009      4.0  
340       3400.0  female  2009      3.4  
341       3775.0    male  2009      3.8  
342       4100.0    male  2009      4.1  
343       3775.0  female  2009      3.8  

[124 rows x 9 columns]

In [12]:
penguins["bill_length_mm"].isnull()

0      False
1      False
2      False
3       True
4      False
       ...  
339    False
340    False
341    False
342    False
343    False
Name: bill_length_mm, Length: 344, dtype: bool

In [13]:
penguins["species"].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

In [14]:
Adelie_pd = penguins[penguins["species"] == 'Adelie']
Gentoo_pd = penguins[penguins["species"] == 'Gentoo']
Crinstrap_pd = penguins[penguins["species"] == 'Chinstrap']

In [15]:
Adelie_pd

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
3    Adelie  Torgersen             NaN            NaN                NaN   
4    Adelie  Torgersen            36.7           19.3              193.0   
..      ...        ...             ...            ...                ...   
147  Adelie      Dream            36.6           18.4              184.0   
148  Adelie      Dream            36.0           17.8              195.0   
149  Adelie      Dream            37.8           18.1              193.0   
150  Adelie      Dream            36.0           17.1              187.0   
151  Adelie      Dream            41.5           18.5              201.0   

     body_mass_g     sex  year  mass_kg  
0         3750.0    male  2007      3.8  
1         3800.0  female  2007      3.8  
2         3250.0  female  2007      3.2  
3            NaN     NaN  2007      NaN  
4         3450.0  female  2007      3.4  
..           ...     ...   ...      ...  
147       3475.0  female  2009      3.5  
148       3450.0  female  2009      3.4  
149       3750.0    male  2009      3.8  
150       3700.0  female  2009      3.7  
151       4000.0    male  2009      4.0  

[152 rows x 9 columns]

In [16]:
Gentoo_pd

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152  Gentoo  Biscoe            46.1           13.2              211.0   
153  Gentoo  Biscoe            50.0           16.3              230.0   
154  Gentoo  Biscoe            48.7           14.1              210.0   
155  Gentoo  Biscoe            50.0           15.2              218.0   
156  Gentoo  Biscoe            47.6           14.5              215.0   
..      ...     ...             ...            ...                ...   
271  Gentoo  Biscoe             NaN            NaN                NaN   
272  Gentoo  Biscoe            46.8           14.3              215.0   
273  Gentoo  Biscoe            50.4           15.7              222.0   
274  Gentoo  Biscoe            45.2           14.8              212.0   
275  Gentoo  Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  year  mass_kg  
152       4500.0  female  2007      4.5  
153       5700.0    male  2007      5.7  
154       4450.0  female  2007      4.4  
155       5700.0    male  2007      5.7  
156       5400.0    male  2007      5.4  
..           ...     ...   ...      ...  
271          NaN     NaN  2009      NaN  
272       4850.0  female  2009      4.8  
273       5750.0    male  2009      5.8  
274       5200.0  female  2009      5.2  
275       5400.0    male  2009      5.4  

[124 rows x 9 columns]

In [17]:
Crinstrap_pd

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  mass_kg  
276       3500.0  female  2007      3.5  
277       3900.0    male  2007      3.9  
278       3650.0    male  2007      3.6  
279       3525.0  female  2007      3.5  
280       3725.0    male  2007      3.7  
..           ...     ...   ...      ...  
339       4000.0    male  2009      4.0  
340       3400.0  female  2009      3.4  
341       3775.0    male  2009      3.8  
342       4100.0    male  2009      4.1  
343       3775.0  female  2009      3.8  

[68 rows x 9 columns]

In [18]:
penguins[(penguins["species"] == 'Chinstrap') & (Crinstrap_pd["body_mass_g"] >= 4000)]

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
284  Chinstrap  Dream            46.0           18.9              195.0   
289  Chinstrap  Dream            52.0           18.1              201.0   
291  Chinstrap  Dream            50.5           19.6              201.0   
295  Chinstrap  Dream            49.2           18.2              195.0   
301  Chinstrap  Dream            52.0           19.0              197.0   
305  Chinstrap  Dream            52.8           20.0              205.0   
307  Chinstrap  Dream            54.2           20.8              201.0   
309  Chinstrap  Dream            51.0           18.8              203.0   
313  Chinstrap  Dream            52.0           20.7              210.0   
315  Chinstrap  Dream            53.5           19.9              205.0   
321  Chinstrap  Dream            50.8           18.5              201.0   
323  Chinstrap  Dream            49.0           19.6              212.0   
329  Chinstrap  Dream            50.7           19.7              203.0   
333  Chinstrap  Dream            49.3           19.9              203.0   
339  Chinstrap  Dream            55.8           19.8              207.0   
342  Chinstrap  Dream            50.8           19.0              210.0   

     body_mass_g     sex  year  mass_kg  
284       4150.0  female  2007      4.2  
289       4050.0    male  2007      4.0  
291       4050.0    male  2007      4.0  
295       4400.0    male  2007      4.4  
301       4150.0    male  2007      4.2  
305       4550.0    male  2008      4.6  
307       4300.0    male  2008      4.3  
309       4100.0    male  2008      4.1  
313       4800.0    male  2008      4.8  
315       4500.0    male  2008      4.5  
321       4450.0    male  2009      4.4  
323       4300.0    male  2009      4.3  
329       4050.0    male  2009      4.0  
333       4050.0    male  2009      4.0  
339       4000.0    male  2009      4.0  
342       4100.0    male  2009      4.1

In [19]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
mass_kg               2
dtype: int64

# How to remove null values

In [20]:
penguins["bill_length_mm"].mean()

np.float64(43.9219298245614)

In [21]:
penguins["bill_length_mm"] = penguins["bill_length_mm"].fillna(penguins["bill_length_mm"].mean())

In [22]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
mass_kg               2
dtype: int64

# If the data is categorical we can use mode()

In [23]:
penguins["sex"] = penguins["sex"].fillna(penguins["sex"].mode()[0])

In [24]:
penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        2
flipper_length_mm    2
body_mass_g          2
sex                  0
year                 0
mass_kg              2
dtype: int64

# Repeat the process of removing the null values to the other columns

In [25]:
penguins["bill_depth_mm"] = penguins["bill_depth_mm"].fillna(penguins["bill_depth_mm"].mean())
penguins["flipper_length_mm"] = penguins["flipper_length_mm"].fillna(penguins["flipper_length_mm"].mean())
penguins["body_mass_g"] = penguins["body_mass_g"].fillna(penguins["body_mass_g"].mean())
penguins["mass_kg"] = penguins["mass_kg"].fillna(penguins["mass_kg"].mean())

# All the null values have been removed

In [26]:
penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
mass_kg              0
dtype: int64

# Changing the data type of the columns

In [27]:
penguins["species"] = penguins["species"].astype("category")
penguins["year"] = penguins["year"].astype("category")
penguins["island"] = penguins["island"].astype("category")
penguins["sex"] = penguins["sex"].astype("category")

In [28]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    category
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      344 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    category
 7   year               344 non-null    category
 8   mass_kg            344 non-null    float64 
dtypes: category(4), float64(5)
memory usage: 15.4 KB


In [29]:
penguins.columns

Index(['species', 'island', 'bill_length_mm', 'bill_depth_mm',
       'flipper_length_mm', 'body_mass_g', 'sex', 'year', 'mass_kg'],
      dtype='object')

# Exploration Data Analytics (EDA)

<h2>1. Which species has the heviest body mass in kg</h2>

In [30]:
Crinstrap_pd["body_mass_g"].mean()

np.float64(3733.0882352941176)

In [31]:
Adelie_pd["body_mass_g"].mean()

np.float64(3700.662251655629)

In [32]:
Gentoo_pd["body_mass_g"].mean()

np.float64(5076.016260162602)

<h1>Insight #1</h1>
<h1>The heaviest species from the dataset is the GENTOO species</h1>

# 7 Questions activity

<hr>

# 1 What species has the highest population?

In [33]:
penguins["species"].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

<h2>Results:</h2>
<li>Adelie has the highest population in compared to the rest</li>
<li>Chinstrap has the lowest population</li>
<hr>

# 2 What species occurs more often in every island?

<h2>The Adelia species in each island</h2>

In [34]:
Adelie = penguins[penguins["species"] == "Adelie"]
Adelie["island"].value_counts()

island
Dream        56
Torgersen    52
Biscoe       44
Name: count, dtype: int64

<h2>The Chinstrap species in each island</h2>

In [35]:
Chinstrap = penguins[penguins["species"] == "Chinstrap"]
Chinstrap["island"].value_counts()

island
Dream        68
Biscoe        0
Torgersen     0
Name: count, dtype: int64

<h2>The Gentoo species in each island</h2>

In [36]:
Gentoo = penguins[penguins["species"] == "Gentoo"]
Gentoo["island"].value_counts()

island
Biscoe       124
Dream          0
Torgersen      0
Name: count, dtype: int64

<h2>Results:</h2>
<li>Adelie has the highest population in Dream island</li>
<li>Chinstrap has the highest population in Dream island since it is the only species there</li>
<li>Gentoo has the highest population in Biscoe island since it is the only species there</li>
<hr>

# 3 What island has the most Crinstrap?

In [37]:
penguins["island"].value_counts().sum()

np.int64(344)

<h2>Chinstrap in Dream island</h2>

In [38]:
Chinstrap_1 = penguins[penguins["species"] == "Chinstrap"]
common_Chinstrap_1 = Chinstrap_1[Chinstrap_1["island"] == "Dream"]  
common_Chinstrap_1["species"].value_counts()

species
Chinstrap    68
Adelie        0
Gentoo        0
Name: count, dtype: int64

<h2>Chinstrap in Torgersen island</h2>

In [39]:
Chinstrap_2 = penguins[penguins["species"] == "Chinstrap"]
common_Chinstrap_2 = Chinstrap_2[Chinstrap_2["island"] == "Torgersen"]  
common_Chinstrap_2["species"].value_counts()

species
Adelie       0
Chinstrap    0
Gentoo       0
Name: count, dtype: int64

<h2>Chinstrap in Biscoe island</h2>

In [40]:
Chinstrap_3 = penguins[penguins["species"] == "Chinstrap"]
common_Chinstrap_3 = Chinstrap_3[Chinstrap_3["island"] == "Biscoe"]  
common_Chinstrap_3["species"].value_counts()

species
Adelie       0
Chinstrap    0
Gentoo       0
Name: count, dtype: int64

<h3>Results:</h3>
<li>We found out that there are no Chinstrap in Torgersen and Biscoe island</li>
<li>We found out that Chinstrap is abundant in Dream island</li>
<hr>

# 4 What is the average size of flippers for all the species?

In [41]:
Chinstrap = penguins[penguins["species"] == "Chinstrap"]["flipper_length_mm"].mean()
Adelie = penguins[penguins["species"] == "Adelie"]["flipper_length_mm"].mean()
Gentoo = penguins[penguins["species"] == "Gentoo"]["flipper_length_mm"].mean()

print("The average flipper size of the Chinstrap species", round(Chinstrap,1))
print("The average flipper size of the Adelie species", round(Adelie,1))
print("The average flipper size of the Gentoo species", round(Gentoo,1))

The average flipper size of the Chinstrap species 195.8
The average flipper size of the Adelie species 190.0
The average flipper size of the Gentoo species 217.1


<h2>Results:</h2>
<li>The average size of flippers for each species range from 190 to 217 in lenght</li>
<li>Gentoo has the largest average size compare to the other species</li>
<li>Adelie has the smallest average size compared to the other species</li>
<hr>

# 5 What island has the most occuring penguin with the weight over 4.5 Kg above?

<h2>The weight of penguins above 4.5 Kg in Dream island</h2>

In [42]:
dream_mass = penguins[(penguins["island"] == "Dream") & (penguins["mass_kg"] >= 4.5)]
dream_mass["island"].value_counts()

island
Dream        6
Biscoe       0
Torgersen    0
Name: count, dtype: int64

<h2>The weight of penguins above 4.5 Kg in Torgersen island</h2>

In [43]:
Torgersen_mass = penguins[(penguins["island"] == "Torgersen") & (penguins["mass_kg"] >= 4.5)]
Torgersen_mass["island"].value_counts()

island
Torgersen    3
Biscoe       0
Dream        0
Name: count, dtype: int64

<h2>The weight of penguins above 4.5 Kg in Biscoe island</h2>

In [44]:
Biscoe_mass = penguins[(penguins["island"] == "Biscoe") & (penguins["mass_kg"] >= 4.5)]
Biscoe_mass["island"].value_counts()

island
Biscoe       110
Dream          0
Torgersen      0
Name: count, dtype: int64

<h2>Results:</h2>
<li>Biscoe island has the most amount of penguins above 4.5 Kg</li>
<li>Torgersen has the lowest among the other islands</li>
<hr>

# 6 how many penguins are there in dream island?

In [45]:
dream_penguins = penguins[penguins["island"] == "Dream"]
species = dream_penguins["species"].value_counts()
total = species.sum()

print(species)
print("Total penguins in Dream island", total)

species
Chinstrap    68
Adelie       56
Gentoo        0
Name: count, dtype: int64
Total penguins in Dream island 124


<h2>Results:</h2>
<li>There are no Gentoo in Dream island</li>
<li>There are a total of 124 penguins in Dream island</li>
<hr>

# 7 What is the most common sex on each island??

<h2>Number of males and females on Dream Island</h2>

In [46]:
dream_penguins = penguins[penguins["island"] == "Dream"]
dream_penguins["sex"].value_counts()

sex
male      63
female    61
Name: count, dtype: int64

<h2>Number of males and females on Torgersen Island</h2>

In [47]:
dream_penguins = penguins[penguins["island"] == "Torgersen"]
dream_penguins["sex"].value_counts()

sex
male      28
female    24
Name: count, dtype: int64

<h2>Number of males and females on Biscoe Island</h2>

In [48]:
dream_penguins = penguins[penguins["island"] == "Biscoe"]
dream_penguins["sex"].value_counts()

sex
male      88
female    80
Name: count, dtype: int64

<h2>Results:</h2>
<li>There are more male penguins in each islands</li>
<li>Biscoe island has the highest male count compare to the onther islands</li>
<li>Torgersen island has the lowest male count among the other islands</li>
<hr>

# DATA AGGREGATION


<H1>Ilang species per island and nakatira?</H1>

In [49]:
PerIsland = penguins.groupby(["species", "island", "sex"]).size()
PerIsland

species    island     sex   
Adelie     Biscoe     female    22
                      male      22
           Dream      female    27
                      male      29
           Torgersen  female    24
                      male      28
Chinstrap  Biscoe     female     0
                      male       0
           Dream      female    34
                      male      34
           Torgersen  female     0
                      male       0
Gentoo     Biscoe     female    58
                      male      66
           Dream      female     0
                      male       0
           Torgersen  female     0
                      male       0
dtype: int64

In [50]:
mass_species = penguins.groupby(['species', 'body_mass_g']).size()
mass_species


species  body_mass_g
Adelie   2700.0         0
         2850.0         2
         2900.0         3
         2925.0         1
         2975.0         1
                       ..
Gentoo   5850.0         3
         5950.0         2
         6000.0         2
         6050.0         1
         6300.0         1
Length: 285, dtype: int64

# More questions

<h1>How manny female penguins are there in in each island</h1>

# TESTING CORRRELATION

<h1>Person R Correlation</h1>
<hr>
<p>Used for testing correlation between two <b>NUMEBRICAL VALUES</b></p>

In [51]:
weight_flipper_cor = penguins["body_mass_g"].corr(penguins['flipper_length_mm'])
weight_flipper_cor

np.float64(0.8712017673060114)

# NUMERICAL TO CATEGORICAL CORRELATION TEST

<H1><b>Library -SciPy (Scientific Python)</b></H1>
<hr>
<h2>from scipy.stats import for oneway</h2>

In [52]:
from scipy.stats import f_oneway 

In [53]:
penguins[penguins['sex'] == 'male']['body_mass_g']
# or use this

0      3750.000000
3      4201.754386
5      3650.000000
7      4675.000000
8      3475.000000
          ...     
334    3800.000000
336    3950.000000
339    4000.000000
341    3775.000000
342    4100.000000
Name: body_mass_g, Length: 179, dtype: float64

In [54]:
x_df = penguins[penguins['sex'] == 'male']
x_df['body_mass_g']

0      3750.000000
3      4201.754386
5      3650.000000
7      4675.000000
8      3475.000000
          ...     
334    3800.000000
336    3950.000000
339    4000.000000
341    3775.000000
342    4100.000000
Name: body_mass_g, Length: 179, dtype: float64

In [55]:
f_oneway(penguins[penguins['sex'] == 'male']['body_mass_g'],penguins[penguins['sex'] == 'female']['body_mass_g'])

F_onewayResult(statistic=np.float64(68.38476171974526), pvalue=np.float64(3.0226600487050246e-15))

In [56]:
result, pvalue = f_oneway(penguins[penguins['sex'] == 'male']['body_mass_g'],penguins[penguins['sex'] == 'female']['body_mass_g'])
pvalue

np.float64(3.0226600487050246e-15)

In [57]:
f_oneway(penguins[penguins['species'] == 'Adelie']['body_mass_g'],
         penguins[penguins['species'] == 'Chinstrap']['body_mass_g'],
         penguins[penguins['species'] == 'Gentoo']['body_mass_g'])

F_onewayResult(statistic=np.float64(338.57291007822204), pvalue=np.float64(1.0069770504490732e-81))

<h1>12 Insights</h1>

<hr>
<h1>1. Is there a correlation between species and their body mass?</h1>

In [58]:
f_oneway(penguins[penguins['species'] == 'Adelie']['body_mass_g'],
                         penguins[penguins['species'] == 'Chinstrap']['body_mass_g'],
                        penguins[penguins['species'] == 'Gentoo']['body_mass_g']).pvalue

np.float64(1.0069770504490732e-81)

<h1>Insights:</h1>
<li>There is a correlation between species and body mass</li>

<hr>
<h1>2. Is there a correlation between their Island and flipper length?</h1>

In [59]:
f_oneway(penguins[penguins['island'] == "Torgersen"]['flipper_length_mm'],
        penguins[penguins['island'] == "Dream"]['flipper_length_mm'],
        penguins[penguins['island'] == "Biscoe"]['flipper_length_mm']).pvalue

np.float64(2.388459860537622e-35)

<h1>Insights:</h1>
<li> There is a correlation between where they live to there flipper lenght</li>

<hr>
<h1>3. Is there a significant difference in bill length between penguins on Biscoe Island with body mass greater than or equal to the island’s average and those below it?</h1>

In [60]:
Biscoe_body_mass_ave = penguins[penguins['island'] == 'Biscoe']['mass_kg'].mean()
Biscoe_body_mass_ave

np.float64(4.71489313561682)

In [73]:
greater_average_penguins = penguins[(penguins['island'] == 'Biscoe') & (penguins['mass_kg'] >= Biscoe_body_mass_ave)]['bill_length_mm']

below_average_penguins = penguins[(penguins['island'] == 'Biscoe') & (penguins['mass_kg'] <= Biscoe_body_mass_ave)]['bill_length_mm']

result = f_oneway(greater_average_penguins, below_average_penguins).pvalue

result

np.float64(6.406911210966658e-26)

<h1>Insights:</h1>
<li>There is a significant difference between body mass and bill length among penguins in Biscoe island </li>
<hr>

# 4. Is there a relation between Islands and the average weight of penguins?

In [88]:
Biscoe_body_mass_ave = penguins[penguins['island'] == 'Biscoe']['mass_kg']
Torgersen_body_mass_ave = penguins[penguins['island'] == 'Torgersen']['mass_kg']
Dream_body_mass_ave = penguins[penguins['island'] == 'Dream']['mass_kg']

result_islands = f_oneway(Biscoe_body_mass_ave , Torgersen_body_mass_ave , Dream_body_mass_ave).pvalue
result_islands

np.float64(1.1793733329967104e-37)

<h1>Insights:</h1>
<li>The result shows that there is a relation between the average weight of various penguins from different islands</li>
<hr>

# 5. Is there a relation between Species and the average weight of penguins?

In [87]:
Adelie_body_mass_ave = penguins[penguins['species'] == 'Adelie']['mass_kg']
Chinstrap_body_mass_ave = penguins[penguins['species'] == 'Chinstrap']['mass_kg']
Gentoo_body_mass_ave = penguins[penguins['species'] == 'Gentoo']['mass_kg']

result_species = f_oneway(Adelie_body_mass_ave , Chinstrap_body_mass_ave , Gentoo_body_mass_ave).pvalue
result_species

np.float64(2.443218973028842e-81)

<h1>Insights:</h1>
<li>The result shows that there is a relation between the average weight of various penguins from different species</li>
<hr>

# 6. Has the the average flipper length of the penguins changed from 2007, 2008 and 2009 the years?

In [100]:
mass_ave_2007 = penguins[penguins['year'] == 2007]['flipper_length_mm']
mass_ave_2008 = penguins[penguins['year'] == 2008]['flipper_length_mm']
mass_ave_2009 = penguins[penguins['year'] == 2009]['flipper_length_mm']

flipper_result = f_oneway(mass_ave_2007 , mass_ave_2008 , mass_ave_2009).pvalue
flipper_result

np.float64(0.0012761391137243161)

<h1>Insights:</h1>
<li>There is no significant change over the years from the size of their flipper length</li>
<hr>

# 7. Do male and female penguins differ significantly from their bill depth? 

In [101]:
male_bill_depth = penguins[penguins['sex'] == "male"]['bill_depth_mm']
female_bill_depth = penguins[penguins['sex'] == "female"]['bill_depth_mm']

sex_result = f_oneway(male_bill_depth, female_bill_depth).pvalue
sex_result

np.float64(1.2906481751545917e-11)

<h1>Insights:</h1>
<li>There is a significant change from male and female penguins when it comes with there sex</li>
<hr>

# 8. Do male penguins from the year 2007 differ significantly from the year 2009 in body mass? 

In [116]:
male_penguins_2007 = penguins[(penguins['sex'] == "male") & (penguins['year'] == 2007)]['mass_kg']
male_penguins_2009 = penguins[(penguins['sex'] == "male") & (penguins['year'] == 2009)]['mass_kg']

result = f_oneway(male_penguins_2007, male_penguins_2009).pvalue
result

np.float64(0.3711930759025889)

<h1>Insights:</h1>
<li>There was a significant change from male penguins in the year 2007 and 2009 when it comes with their body mass</li>
<hr>

# 9. Does male penguins bill length and flipper length correlate to one another?

In [120]:
male_penguins = penguins[penguins['sex'] == 'male']

male_result = male_penguins["bill_length_mm"].corr(male_penguins['flipper_length_mm'])
male_result

np.float64(0.6711843498471304)

<h1>Insights:</h1>
<li>Bill length and flipper lenght do correlate to one another when it comes to male penguins</li>
<hr>

# 10. Does female penguins bill length and flipper length correlate to one another?

In [121]:
female_penguins = penguins[penguins['sex'] == 'female']

female_result = female_penguins["bill_length_mm"].corr(female_penguins['flipper_length_mm'])
female_result

np.float64(0.5714736752416979)

<h1>Insights:</h1>
<li>Bill length and flipper lenght do correlate to one another when it comes to female penguins</li>
<hr>

# 11. Do female and male penguins differ to one another when it comes to their bill length and flipper length?

In [133]:

male_bill = male_penguins['bill_length_mm']
female_bill = female_penguins['bill_length_mm']

final_result = f_oneway(male_bill, female_bill).pvalue
final_result

np.float64(9.307501289283404e-10)

In [134]:
male_flipper = male_penguins['flipper_length_mm']
female_flipper = female_penguins['flipper_length_mm']

final_result = f_oneway(male_flipper, female_flipper).pvalue
final_result

np.float64(4.88065474065626e-06)

<h1>Insights:</h1>
<li>The results shows that there is a significant difference between their flipper length and bill length from male and female </li>
<hr>

# 12. Does the body mass and bill depth correlate to one another?

In [140]:

result = penguins["body_mass_g"].corr(penguins['bill_depth_mm'])
result

np.float64(-0.47191562118606645)

<h1>Insights:</h1>
<li>It seems like their are penguins with a huge body mass but they have small bills </li>
<hr>

In [92]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen        39.10000       18.70000         181.000000   
1       Adelie  Torgersen        39.50000       17.40000         186.000000   
2       Adelie  Torgersen        40.30000       18.00000         195.000000   
3       Adelie  Torgersen        43.92193       17.15117         200.915205   
4       Adelie  Torgersen        36.70000       19.30000         193.000000   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream        55.80000       19.80000         207.000000   
340  Chinstrap      Dream        43.50000       18.10000         202.000000   
341  Chinstrap      Dream        49.60000       18.20000         193.000000   
342  Chinstrap      Dream        50.80000       19.00000         210.000000   
343  Chinstrap      Dream        50.20000       18.70000         198.000000   

     body_mass_g     sex  year   mass_kg  
0    3750.000000    male  2007  3.800000  
1    3800.000000  female  2007  3.800000  
2    3250.000000  female  2007  3.200000  
3    4201.754386    male  2007  4.202047  
4    3450.000000  female  2007  3.400000  
..           ...     ...   ...       ...  
339  4000.000000    male  2009  4.000000  
340  3400.000000  female  2009  3.400000  
341  3775.000000    male  2009  3.800000  
342  4100.000000    male  2009  4.100000  
343  3775.000000  female  2009  3.800000  

[344 rows x 9 columns]